HumanResourceMachine/Services/BusinessLogicService.cs

```diff
@@ -0,0 +1,57 @@
+﻿using HumanResourceMachine.Context;
+using HumanResourceMachine.Entities;
+using HumanResourceMachine.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace HumanResourceMachine.Services
+{
+    public class BusinessLogicService : IBusinessLogicService
+    {
+        private readonly HRMContext _context;
+
+        public BusinessLogicService(HRMContext context)
+        {
+            _context = context;
+        }
+
+        public IEnumerable<Human> GetAllHumans()
+        {
+            return _context.People.ToList();
+        }
+
+        public Human GetHumanById(int id)
+        {
+            var result = _context.People.Find(id);
+            if (result is null)
+            {
+                throw new ArgumentNullException("Object doesn't exist.", nameof(result));
+            }
+
+            return result;
+        }
+
+        public void AddHuman(Human human)
+        {
+            _context.People.Add(human);
+            _context.SaveChanges();
+        }
+
+        public void UpdateHuman(Human human)
+        {
+            _context.Entry(human).State = EntityState.Modified;
```

Comment:
why not update()? i would say it is better to use

Reply:
Because update method can create new Entity if it's doesn't exist.  This Method should only update property of existing entity.